src/main/java/com/app/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -46,7 +52,17 @@ public UserRegistrationResponseDto register(
                 new NoSuchElementException("Can't find such role: " + RoleName.ROLE_USER));
         roles.add(defoultRole);
         user.setRoles(roles);
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
         User saveUser = userRepository.save(user);
         return userMapper.toUserResponse(saveUser);
     }
+
+    @Override
+    public User getAuthenticatedUser() {

```

Comment:
Just comment: to not write this method you can pass Authentication in controllers methods to services, and there you can extract user from the authentication object: `   //Controller
@GetMapping
    public ShoppingCartDto getShoppingCart(Authentication authentication) {
        return shoppingCartService.getShoppingCart(authentication);
    }`
`//Service
 @Override
    public ShoppingCartDto getShoppingCart(Authentication authentication) {
        User currentUser = (User) authentication.getPrincipal();
        ......
`